### PR TITLE
Animation and transition types

### PR DESCRIPTION
**Pull request by Starzu**

### CODE REVIEW

File: core/src/main/scala/scalacss/Values.scala [hunk 1]
```diff
@@ -47,6 +47,8 @@ object Literal extends TypedLiteralAliases {
   object Typed {
     final val `0` = Length(0, LengthUnit.px)
 
+    class count(n: Double) extends Literal(s"$n")
```
Comment (japgolly) [inline core/src/main/scala/scalacss/Values.scala]: Please change `s"$n"` to `n.toString`. The `s""` comes with overhead we don't need here.

File: core/src/main/scala/scalacss/Attrs.scala [hunk 2]
```diff
@@ -2126,7 +2156,18 @@ object Attrs {
...
+    override val attr: Attr = Attr.real("transition-timing-function", Transform keys CanIUse.transitions)
+    @inline def cubicBezier(x1: Double, y1: Double, x2: Double, y2: Double) = avl(new LT.CubicBezier(x1, y1, x2, y2))
+    @inline def steps(steps: Int, direction: LT.TimingFunctionDirection) = avl(new LT.Steps(steps, direction))
+    @inline def linear = avl(LT.linear)
+    @inline def ease = avl(LT.ease)
+    @inline def easeIn = avl(LT.easeIn)
+    @inline def easeInOut = avl(LT.easeInOut)
+    @inline def easeOut = avl(LT.easeOut)
+    @inline def stepStart = avl(LT.stepStart)
+    @inline def stepEnd = avl(LT.stepEnd)
```
Comment (japgolly) [inline core/src/main/scala/scalacss/Attrs.scala]: Here (and the groups above), please align all the whitespace so all the `=` are aligned. Big improvement of readablity. Does your IDE support multiple/verticle cursors? It's super quick with that feature.

File: core/src/main/scala/scalacss/Values.scala [hunk 3]
```diff
@@ -75,6 +77,22 @@ object Literal extends TypedLiteralAliases {
...
+    sealed trait TimingFunction
+    class CubicBezier(x1: Double, y1: Double, x2: Double, y2: Double) extends Literal(s"cubic-bezier($x1, $y1, $x2, $y2)") with TimingFunction
+    class Steps(steps: Int, direction: TimingFunctionDirection) extends Literal(s"steps($steps, ${direction.value})") with TimingFunction
+    object linear extends Literal("linear") with TimingFunction
+    object ease extends Literal("ease") with TimingFunction
+    object easeIn extends Literal("ease-in") with TimingFunction
+    object easeInOut extends Literal("ease-in-out") with TimingFunction
+    object easeOut extends Literal("ease-out") with TimingFunction
+    object stepStart extends Literal("step-start") with TimingFunction
+    object stepEnd extends Literal("step-end") with TimingFunction
```
Comment (japgolly) [inline core/src/main/scala/scalacss/Values.scala]: Please format whitespace so all `extends` are aligned like other blocks.

File: core/src/main/scala/scalacss/Values.scala [hunk 4]
```diff
@@ -75,6 +77,22 @@ object Literal extends TypedLiteralAliases {
...
     object outset extends Literal("outset") with BrStyle
+
+    // <timing-function>
+    sealed trait TimingFunctionDirection { def value: Value }
+    object start extends Literal("start") with TimingFunctionDirection
+    object end extends Literal("end") with TimingFunctionDirection
+
+    sealed trait TimingFunction
+    class CubicBezier(x1: Double, y1: Double, x2: Double, y2: Double) extends Literal(s"cubic-bezier($x1, $y1, $x2, $y2)") with TimingFunction
+    class Steps(steps: Int, direction: TimingFunctionDirection) extends Literal(s"steps($steps, ${direction.value})") with TimingFunction
```
Comment (japgolly) [inline core/src/main/scala/scalacss/Values.scala]: Either these two need to be `cubicBezier` & `steps`, or the new `count` above needs to be `Count`.

File: core/src/main/scala/scalacss/Values.scala (hunk 3 above)
Comment (japgolly) [inline core/src/main/scala/scalacss/Values.scala]: Also you'll need to add links to all of these + `start` + `end` to the `TypedLiteralAliases` trait at the top of this file.

**Comment by japgolly** [comment]
Hi. I've added comments to the diff.

In addition could you:
* add a quick test to `InlineTest` or something, to ensure the DSL works and the output is as expected (and that it stays that way).
* once you're done, please squash your commits.

Cheers

File: core/src/main/scala/scalacss/Attrs.scala (hunk 2 above)
Comment (Starzu) [inline core/src/main/scala/scalacss/Attrs.scala]: Yes, it does. Thanks for the hint.

**Comment by japgolly** [comment]
Hey @Starzu, I'm not sure if you're finished working on this but [Travis is failing](https://travis-ci.org/japgolly/scalacss/jobs/101056629). Just let me know when you're ready for me to take another look at this - I don't get notifications when you change a PR's commits.

**Comment by Starzu** [comment]
@japgolly 

The reason of build fail is interesting:
> [info] Fast optimizing /home/travis/build/japgolly/scalacss/core/target/js/scala-2.11/core-test-fastopt.js
> No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.

Whole "test" process takes about 240s on my local machine with JDK8. It is similar on master branch. 

Do you have any idea what is wrong?

**Comment by japgolly** [comment]
@Starzu I just checked out your branch and did a clean test. Finished in 28s without any problems - all tests passed - I don't know what's wrong with Travis.

Ok, I'm heading out now - will merge this tomorrow

**Comment by japgolly** [comment]
Sorry - I forgot to mention this before - could you please also update the [changelog](https://github.com/japgolly/scalacss/blob/master/doc/history/0.4.md) and squash the commit again?

**Comment by Starzu** [comment]
@japgolly Done :)

**Comment by japgolly** [comment]
Thanks1

**Pull request merged**